osu.Game.Rulesets.Osu/Mods/OsuModRandom.cs

```diff
@@ -0,0 +1,364 @@
+            );
+        }
+    }
+
+    public class OsuModRandomSettingsControl : SettingsItem<int>
+    {
+        [Resolved]
+        private static GameHost host { get; set; }
+
+        [BackgroundDependencyLoader]
+        private void load(GameHost gameHost) => host = gameHost;
+
+        protected override Drawable CreateControl() => new SeedControl
+        {
+            RelativeSizeAxes = Axes.X,
+            Margin = new MarginPadding { Top = 5 }
+        };
+
+        private sealed class SeedControl : CompositeDrawable, IHasCurrentValue<int>
+        {
+            private readonly BindableWithCurrent<int> current = new BindableWithCurrent<int>();
+
+            public Bindable<int> Current
+            {
+                get => current;
+                set => Scheduler.Add(() => current.Current = value);
+            }
+
+            private readonly OsuNumberBox seedNumberBox;
+
+            public SeedControl()
+            {
+                AutoSizeAxes = Axes.Y;
+
+                InternalChildren = new[]
+                {
+                    new GridContainer
+                    {
+                        RelativeSizeAxes = Axes.X,
+                        AutoSizeAxes = Axes.Y,
+                        ColumnDimensions = new[]
+                        {
+                            new Dimension(),
+                            new Dimension(GridSizeMode.Absolute, 2),
+                            new Dimension(GridSizeMode.Relative, 0.25f)
+                        },
+                        RowDimensions = new[]
+                        {
+                            new Dimension(GridSizeMode.AutoSize)
+                        },
+                        Content = new[]
+                        {
+                            new Drawable[]
+                            {
+                                seedNumberBox = new OsuNumberBox
+                                {
+                                    RelativeSizeAxes = Axes.X,
+                                    CommitOnFocusLost = true
+                                },
+                                null,
+                                new TriangleButton
+                                {
+                                    RelativeSizeAxes = Axes.Both,
+                                    Height = 1,
+                                    Text = "Copy",
+                                    Action = copySeedToClipboard
+                                }
+                            }
+                        }
+                    }
+                };
+
+                seedNumberBox.Current.BindValueChanged(onTextBoxValueChanged);
+            }
+
+            private void onTextBoxValueChanged(ValueChangedEvent<string> e)
+            {
+                string seed = e.NewValue;
+
+                while (!string.IsNullOrEmpty(seed) && !int.TryParse(seed, out _))
+                    seed = seed[..^1];
```

Comment:
This should not be a thing. The number box already validates characters. At the very most I'd only trim whitespace.